vault/provider.go

```diff
@@ -634,6 +640,11 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 		return nil, errors.New("no vault token found")
 	}
 
+	tokenName := d.Get("token_name").(string)
+	if tokenName == "" {
```

Comment:
This will never be an empty string if the default func on line 62 defaults to "terraform". That's why I proposed the change above. That way it'll be like:
- Is it in the env variable? If so use that.
- Is it set as token_name? If so use that.
- Otherwise, use "terraform".

Reply:
Thanks, you are right, the "terraform" fallback/default is duplicated.

By adding the Acc tests, I discovered, that the token name was empty, if token_name is defined as empty string "". That's why I added this fallback.

I will change it to your proposal and I will enhance documentation regarding default value.

Reply:
Done